oauth2client/client.py

```diff
@@ -554,6 +554,11 @@ def new_request(uri, method='GET', body=None, headers=None,
         else:
           headers['user-agent'] = self.user_agent
 
+      body_stream_position = None
+      if all(hasattr(body, stream_prop) for stream_prop in ('read',
```

Comment:
See http://doc.bazaar.canonical.com/developers/code-style.html#hasattr-and-getattr for a brief lesson on why hasattr is fundamentally broken and should never be used.